libraries/AP_Networking/AP_Networking.cpp

```diff
@@ -0,0 +1,393 @@
+    // @RebootRequired: True
+    // @User: Advanced
+    AP_GROUPINFO("GWADDR2", 9,  AP_Networking,    _param.gwaddr[2],   AP_NETWORKING_DEFAULT_STATIC_GW_ADDR2),
+
+    // @Param: GWADDR3
+    // @DisplayName: Gateway IP Address LSB
+    // @Description: Allows setting static GW address
+    // @Range: 0 255
+    // @RebootRequired: True
+    // @User: Advanced
+    AP_GROUPINFO("GWADDR3", 10,  AP_Networking,    _param.gwaddr[3],   AP_NETWORKING_DEFAULT_STATIC_GW_ADDR3),
+
+    // @Param: MACADDR0
+    // @DisplayName: MAC Address MSbyte
+    // @Description: Allows setting MAC address
+    // @Range: 0 255
+    // @RebootRequired: True
+    // @User: Advanced
+    AP_GROUPINFO("MACADDR0", 11,  AP_Networking,    _param.macaddr[0],  AP_NETWORKING_DEFAULT_MAC_ADDR0),
+
+    // @Param: MACADDR1
+    // @DisplayName: MAC Address 2nd byte
+    // @Description: Allows setting MAC address
+    // @Range: 0 255
+    // @RebootRequired: True
+    // @User: Advanced
+    AP_GROUPINFO("MACADDR1", 12,  AP_Networking,    _param.macaddr[1],   AP_NETWORKING_DEFAULT_MAC_ADDR1),
+
+    // @Param: MACADDR2
+    // @DisplayName: MAC Address 3rd byte
+    // @Description: Allows setting MAC address
+    // @Range: 0 255
+    // @RebootRequired: True
+    // @User: Advanced
+    AP_GROUPINFO("MACADDR2", 13,  AP_Networking,    _param.macaddr[2],   AP_NETWORKING_DEFAULT_MAC_ADDR2),
+
+    // @Param: MACADDR3
+    // @DisplayName: MAC Address 4th byte
+    // @Description: Allows setting MAC address
+    // @Range: 0 255
+    // @RebootRequired: True
+    // @User: Advanced
+    AP_GROUPINFO("MACADDR3", 14,  AP_Networking,    _param.macaddr[3],   AP_NETWORKING_DEFAULT_MAC_ADDR3),
+
+    // @Param: MACADDR4
+    // @DisplayName: MAC Address 5th byte
+    // @Description: Allows setting MAC address
+    // @Range: 0 255
+    // @RebootRequired: True
+    // @User: Advanced
+    AP_GROUPINFO("MACADDR4", 15,  AP_Networking,    _param.macaddr[4],   AP_NETWORKING_DEFAULT_MAC_ADDR4),
+
+    // @Param: MACADDR5
+    // @DisplayName: MAC Address LSb
+    // @Description: Allows setting MAC address
+    // @Range: 0 255
+    // @RebootRequired: True
+    // @User: Advanced
+    AP_GROUPINFO("MACADDR5", 16,  AP_Networking,    _param.macaddr[5],   AP_NETWORKING_DEFAULT_MAC_ADDR5),
+
+    AP_GROUPEND
+};
+
+AP_Networking::AP_Networking(void)
+{
+#if CONFIG_HAL_BOARD == HAL_BOARD_SITL
+    if (_singleton != nullptr) {
+        AP_HAL::panic("AP_Networking must be singleton");
+    }
+#endif
+
+    _singleton = this;
+    AP_Param::setup_object_defaults(this, var_info);
+}
+
+
+void AP_Networking::init()
+{
+#ifdef HAL_GPIO_ETH_ENABLE
+    hal.gpio->pinMode(HAL_GPIO_ETH_ENABLE, HAL_GPIO_OUTPUT);
+    hal.gpio->write(HAL_GPIO_ETH_ENABLE, _param.enabled ? 1 : 0);
+#endif
+
+    if (!_param.enabled || _init.done) {
+        return;
+    }
+
+    // set default MAC Address lower 3 bytes to UUID if possible
+    uint8_t uuid[12];
+    uint8_t uuid_len = sizeof(uuid);
```

Comment:
Add const?

Reply:
Sure thing

Reply:
ah! It cannot be const because get_system_id_unformatted() populates the len